internal/storage/sql/postgres/postgres.go

```diff
@@ -147,6 +147,25 @@ func (s *Store) CreateConstraint(ctx context.Context, r *flipt.CreateConstraintR
 	return constraint, nil
 }
 
+func (s *Store) CreateRollout(ctx context.Context, r *flipt.CreateRolloutRequest) (*flipt.Rollout, error) {
+	rollout, err := s.Store.CreateRollout(ctx, r)
+
+	if err != nil {
+		var perr *pq.Error
+
+		if errors.As(err, &perr) && perr.Code.Name() == constraintForeignKeyErr {
+			if segment := r.GetSegment(); segment != nil {
+				return nil, errs.ErrNotFoundf(`flag "%s/%s or segment %s"`, r.NamespaceKey, r.FlagKey, segment.SegmentKey)
+			}
+			return nil, errs.ErrNotFoundf(`flag "%s/%s"`, r.NamespaceKey, r.FlagKey)
```

Comment:
I think we should be able to better glean the reason for this error from the metadata on the `*pq.Error` itself (same for the MySQL error). As opposed to assuming not found here when a segment has been supplied.

Take for example the case where the segment exists, but the flag does not. I think in this case it would say segment not found?

I believe there are a number of fields on the errors that might give more accurate detail.
e.g. `perr.Constraint` or `perr.Table`:
https://github.com/lib/pq/blob/3d613208bca2e74f2a20e04126ed30bcb5c4cc27/error.go#L36C25-L39

I would probably try and reverse engineer from what the error currently says. I would put in a log statement temporarily to see what is in the more detailed error (`fmt.Printf("%#v\n", perr)`) and see if we can be more deterministic on the cause from these fields on the error.

Reply:
We discussed outside that this kind of improvement is needed in a broader cross-section of the API.
So we would forgot it for now and make an issue to capture that.